From dceb46116a131564c1cfa7f069fd1a822879436f Mon Sep 17 00:00:00 2001
From: Clayton Shotwell <clayton.shotwell@rockwellcollins.com>
Date: Thu, 26 Mar 2015 12:33:10 -0500
Subject: [PATCH] Make zos-remote plugin optional

The zos-remote plugin is meant to use LDAP authentication to verify a
remote audit user. Add the ability to disable the plugin if the feature
is not desired.

Signed-off-by: Clayton Shotwell <clayton.shotwell@rockwellcollins.com>
---
 audisp/plugins/Makefile.am |  6 ++++--
 configure.ac               | 10 ++++++++++
 2 files changed, 14 insertions(+), 2 deletions(-)

diff --git a/audisp/plugins/Makefile.am b/audisp/plugins/Makefile.am
index b0fa60a..2cba14b 100644
--- a/audisp/plugins/Makefile.am
+++ b/audisp/plugins/Makefile.am
@@ -22,8 +22,10 @@
 
 CONFIG_CLEAN_FILES = *.loT *.rej *.orig
 
-SUBDIRS = builtins zos-remote remote
-#SUBDIRS = builtins zos-remote
+SUBDIRS = builtins remote
+if ENABLE_ZOS_REMOTE
+SUBDIRS += zos-remote
+endif
 if HAVE_PRELUDE
 SUBDIRS += prelude
 endif
diff --git a/configure.ac b/configure.ac
index f29fa41..2e84b07 100644
--- a/configure.ac
+++ b/configure.ac
@@ -155,6 +155,16 @@ fi
 AM_CONDITIONAL(ENABLE_LISTENER, test "x$enable_listener" != "xno")
 AC_MSG_RESULT($enable_listener)
 
+#audisp zos-remote plugin
+AC_MSG_CHECKING(whether to include audisp ZOS remote plugin)
+AC_ARG_ENABLE(zos-remote,
+	      [AS_HELP_STRING([--disable-zos-remote],
+			      [Disable audisp ZOS remote plugin])],
+	      enable_zos_remote=$enableval,
+	      enable_zos_remote=yes)
+AM_CONDITIONAL(ENABLE_ZOS_REMOTE, test "x$enable_zos_remote" != "xno")
+AC_MSG_RESULT($enable_zos_remote)
+
 #gssapi
 AC_ARG_ENABLE(gssapi_krb5,
 	[AS_HELP_STRING([--enable-gssapi-krb5],[Enable GSSAPI Kerberos 5 support @<:@default=no@:>@])],
-- 
1.9.1

